src/store/modules/selectors.js

```diff
@@ -0,0 +1,47 @@
+const state = {
+  isShown: false,
+  type: "single",
+  options: [],
+  title: "",
+  info: "",
+};
+
+const getters = {
+  isSingleSelectorShown: (state) => {
+    return state.isShown && state.type == "single";
+  },
+};
+
+const actions = {
+  showSelector({ commit }, params) {
+    commit("showSelector", params);
+  },
+  hideSelector({ commit }) {
+    commit("hideSelector");
+  },
+};
+
+const mutations = {
+  showSelector(state, params) {
+    state.isShown = true;
+    state.type = params.type;
+    state.options = params.options;
+
+    if (params.title != undefined) state.title = params.title;
+    if (params.info != undefined) state.info = params.info;
```

Comment:
Why not apply the same checks for type and options as well?

Reply:
And if we check all the keys, we can just paste a deepclone into state from params instead of putting single keys